File: Simulations/R/BIC.R

ts_df <- read.csv("Simulations/Products/data_ts.csv")
u_hat <- residuals(lm(y~X_2,data=ts_df))
lm(y~X_2,data=ts_df) |> summary()
lm(y~X_2,data=ts_df) |> BIC()

my_BIC <- function(resid,k,method){
    n<-length(resid)
    SSR<- sum(resid^2)
    sigma_2 <- var(resid)
    if (method=="SSR") {
        return(n*log(SSR)+(k+1)*log(n)-n*log(n))
    } else {
       return(n*log(sigma_2)+(k+1)*log(n))
    }
}


my_BIC(u_hat,2,"SSR")
my_BIC(u_hat,2,"")
lm(y~X_2,data=ts_df) |> BIC()
lm(y~X_2,data=ts_df) |> AIC()

File: Theta/functions/bic.m
function value = bic(residuals,X,method)
    [n, k] = size(X);
    % SSR=residuals'*residuals;
    sigma_2 = var(residuals);
    switch method
        case 'hansen'
            value = n*log(2*pi*sigma_2)+n+(k+1)*log(n);      
        otherwise
            value = n*log(sigma_2) + k*log(n);
    end
end